Codespell: fixed typos, added config and workflow

yarikoptic opened the pull request:
## Motivation

Still on the mission

## How to test the behavior?

Run codespell

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [ ] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

yarikoptic committed:
Add rudimentary .codespellrc

yarikoptic committed:
Add github action to codespell dev on push and PRs

yarikoptic commented on docs/gallery/general/extensions.py:
```diff
@@ -326,7 +326,7 @@ class PotatoSack(MultiContainerInterface):
 
 # You can add potatoes to a potato sack in different ways
-potato_sack = PotatoSack(potatos=Potato(name='potato1', age=2.3, weight=3.0))
+potato_sack = PotatoSack(potatoes=Potato(name='potato1', age=2.3, weight=3.0))
```
just now mentioned this one -- should I also fix `potatos` above or leave any `potatos` untouched?

rly replied:
Please leave `potatos` untouched. The variable name is autogenerated, which doesn't apply 100% of the time...

yarikoptic replied:
done, left potatos behind.

yarikoptic committed:
Leave potatos alone

yarikoptic committed:
[DATALAD RUNCMD] Run codespell throughout fixing some typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

rly commented:
Thanks @yarikoptic . Could you please update `CHANGELOG.md`? Otherwise it looks good to me.

bendichter committed:
Update CHANGELOG.md

bendichter commented:
@rly added

yarikoptic commented on CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+## PyNWB 2.3.2 (Upcoming)
+
+### Enhancements and minor changes
+- Fixed typos and added codespell GitHub action to check spelling in the future. @yarikoptic [#1648](https://github.com/NeurodataWithoutBorders/pynwb/pull/1648)
```
Thank you @bendichter !

rly commented:
Thanks to you both!
